doc/tutorials/enable_ivshmem.rst

```diff
@@ -0,0 +1,170 @@
+         -s 4,virtio-net,tap0 \
+         -s 6,ivshmem,dm:/test,2097152\
+         -s 7,virtio-rnd \
+         --ovmf /usr/share/acrn/bios/OVMF.fd \
+         $vm_name
+
+
+   - VM2 Launch Script Sample
+
+     .. code-block:: none
+        :emphasize-lines: 5
+
+        acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+         -s 2,pci-gvt -G "$2" \
+         -s 3,virtio-blk,/home/acrn/uos2.img \
+         -s 4,virtio-net,tap0 \
+         -s 5,ivshmem,dm:/test,2097152\
+         --ovmf /usr/share/acrn/bios/OVMF.fd \
+         $vm_name
+
+2. Boot two VMs and use ``lspci | grep "shared memory"`` to verify that the virtual device is ready for each VM.
+
+   -  For VM1, it shows ``00:06.0 RAM memory: Red Hat, Inc. Inter-VM shared memory (rev 01)``
+   -  For VM2, it shows ``00:05.0 RAM memory: Red Hat, Inc. Inter-VM shared memory (rev 01)``
+
+3. As recorded in the `PCI ID Repository <https://pci-ids.ucw.cz/read/PC/1af4>`_,
+   the ``ivshmem`` device vendor ID is ``1af4`` (RedHat) and device ID is ``1110``
+   (Inter-VM shared memory).  Use these commands to probe the device::
+
+     $ sudo modprobe uio
+     $ sudo modprobe uio_pci_generic
+     $ sudo echo "1af4 1110" > /sys/bus/pci/drivers/uio_pci_generic/new_id
+
+.. note:: These commands are applicable to Linux-based guests with ``CONFIG_UIO`` and ``CONFIG_UIO_PCI_GENERIC`` enabled.
+
+4. Finally, a user application can get the shared memory base address from
+   the ``ivshmem`` device BAR resource
+   (``/sys/class/uio/uioX/device/resource2``) and the shared memory size from
+   the ``ivshmem`` device config resource
+   (``/sys/class/uio/uioX/device/config``).
+
+   The ``X`` in ``uioX`` above, is a number that can be retrieved using the
+   ``ls`` command:
+
+   - For VM1 use ``ls -lh /sys/bus/pci/devices/0000:00:06.0/uio``
+   - For VM2 use ``ls -lh /sys/bus/pci/devices/0000:00:05.0/uio``
+
+hv-land example
+===============
+
+This example uses hv-land inter-vm communication between two
+Linux-based VMs (VM0 is a pre-launched VM and VM2 is a post-launched VM).
+
+1. Configure shared memory for the communication between VM0 and VM2 for hybrid_rt
+   scenario on whl-ipc-i5 board, the shared memory name is ``hv:/shm_region_0``,
+   and shared memory size is 2M bytes:
+
+   - Edit XML configuration file for hybrid_rt scenario on whl-ipc-i5 board
+     ``acrn-hypervisor/misc/vm_configs/xmls/config-xmls/whl-ipc-i5/hybrid_rt.xml``
+     to enable ``ivshmem`` and configure the shared memory region using the format
+     ``shm_name, shm_size, VM IDs`` (as described above in the ACRN dm boot parameters).
+     The region name must start with ``hv:/`` for an hv-land shared region, and we'll allocate 2MB
+     shared between VMs 0 and 2:
+
+     .. code-block:: none
+        :emphasize-lines: 2,3
+
+        <IVSHMEM desc="IVSHMEM configuration">
+               <IVSHMEM_ENABLED>y</IVSHMEM_ENABLED>
+               <IVSHMEM_REGION>hv:/shm_region_0, 0x200000, 0:2</IVSHMEM_REGION>
```

Comment:
You use an hex representation for `shm_size` here but used an integer above (for dm-land). Can we use both interchangeably?

Reply:
Sure,  I will use hexadecimal value to describe the size, and the size unit is MB.